common/leaderelection/leaderelection.go

```diff
@@ -179,3 +188,67 @@ func (e *natsEventBusElector) RunOrDie(ctx context.Context, callbacks LeaderCall
 		}
 	}
 }
+
+type kubernetesElector struct {
+	namespace string
+	leasename string
+	hostname  string
+}
+
+func newKubernetesElector(namespace string, leasename string, hostname string) (Elector, error) {
+	return &kubernetesElector{
+		namespace: namespace,
+		leasename: leasename,
+		hostname:  hostname,
+	}, nil
+}
+
+func (e *kubernetesElector) RunOrDie(ctx context.Context, callbacks LeaderCallbacks) {
+	logger := logging.FromContext(ctx)
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		logger.Fatalw("Failed to retrieve kubernetes config", zap.Error(err))
+	}
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		logger.Fatalw("Failed to create kubernetes client", zap.Error(err))
+	}
+
+	lock := &resourcelock.LeaseLock{
+		LeaseMeta: metav1.ObjectMeta{
+			Name:      e.leasename,
+			Namespace: e.namespace,
+		},
+		Client: client.CoordinationV1(),
+		LockConfig: resourcelock.ResourceLockConfig{
+			Identity: e.hostname,
```

Comment:
How to opt in following configuration?

```yaml
  leaseDuration: 15
  renewDeadline: 10
  retryPeriod: 2
```

Reply:
What are your thoughts on adding this to Sensor/EventSource, if omitted we can default to the EventBus-based leader election
```
leaderElection:
  leaseDuration: int
  renewDeadline: int
  retryPeriod: int
```

Reply:
How about keeping it as is, and revisit it later if it's highly demanded for customization?

Reply:
Sounds like a good plan to me!